sisyphus/hash.py

```diff
@@ -4,6 +4,8 @@
 import pathlib
 from inspect import isclass, isfunction, ismemberdescriptor
 
+from sisyphus.job_path import AbstractPath
```

Comment:
Thank you for taking the time of profiling this and additionally for taking the time to contribute the changes to our repo!

I noticed that our tests are failing. `black` can be fixed trivially later, but I think the CI/test might indicate a problem.
We have a .. unorthodox way of importing the `global_settings` so that they can be overridden by user settings in a separate file. This leads to problems with circular imports: in `hash` we now added an import from `job_path` which in turn imports `global_settings` which imports `hash`. And the symptoms can be found in the CI pipeline.
Maybe @critias or @JackTemaki can explain the problem better 🙈 

Maybe moving the import into the `get_object_state` function would already solve it by breaking the loop?